### PR TITLE
fix(menu/ImportIEDs) correct TEMPLATE name for illegal characters

### LINKED ISSUE #1142: Default template IED name generation in Import IEDs results in schema invalid file
**Describe the bug**

Our default way of building the name from the manufacturer and type can readily cause schema invalid projects.

IED name is highly qualified (A-Z,a-z,0-9 and _) but manufacturer and type are "more loose" and especially it is common for the hyphen (-) to be used.

Our default generation then causes a schema invalid file when deciding on the IED name:

https://github.com/openscd/open-scd/blob/2171569a9885cd105804cf342906799e678f8009/src/menu/ImportIEDs.ts#L31-L38

**Expected behavior**

When importing IEDs from an icd file the generated name doesn't cause schema validity errors.

**How to Reproduce**

1. Create an empty project.
2. Attempt to import the ICD file in the attached zip file:
    [XAT_PCS-221S_(18Jan2023FW)_StandardDesign.icd.zip](https://github.com/openscd/open-scd/files/10488446/XAT_PCS-221S_.18Jan2023FW._StandardDesign.icd.zip)

**Screenshots**

image

**Additional context**

We need something like this:

```js
function uniqueTemplateIedName(doc: XMLDocument, ied: Element): string {
  const [manufacturer, type] = ['manufacturer', 'type'].map(attr =>
    ied.getAttribute(attr)?.replace(/[^A-Za-z0-9_]/, '')
  );
  const nameCore =
    manufacturer || type
      ? `${manufacturer ?? ''}${type ? `_${type}` : ''}`
      : 'TEMPLATE_IED';
```

PR in some time, perhaps with a regression test.

### PR DESCRIPTION
**Pull request by danyill**
Closes #1142

### CODE REVIEW

File: test/integration/editors/triggered/ImportIedsPlugin.test.ts [hunk 1]
```diff
@@ -78,6 +78,27 @@ describe('ImportIedsPlugin', () => {
         .exist;
     });
 
+    it('renames TEMPLATE IED element if manufacturer/type has illegal characterss', async () => {
```
Comment (JakobVogelsang) [inline test/integration/editors/triggered/ImportIedsPlugin.test.ts:81]:
```suggestion
    it('renames TEMPLATE IED element if manufacturer/type has illegal characters', async () => {
```

File: test/integration/editors/triggered/ImportIedsPlugin.test.ts [hunk 2]
```diff
@@ -449,5 +470,13 @@ describe('ImportIedsPlugin', () => {
       expect(parent.history[0].kind).to.equal('error');
       expect(parent.history[0].title).to.equal('[import.log.parsererror]');
     });
+
```
Comment (JakobVogelsang) [inline test/integration/editors/triggered/ImportIedsPlugin.test.ts:473]: What is the empty space fore here?

File: test/integration/editors/triggered/ImportIedsPlugin.test.ts [hunk 3]
```diff
@@ -449,5 +470,13 @@ describe('ImportIedsPlugin', () => {
...
+
+
+
   });
+
```
Comment (JakobVogelsang) [inline test/integration/editors/triggered/ImportIedsPlugin.test.ts:478]: Same here:)

**Review by JakobVogelsang — DISMISSED** [review]
LGTM

File: test/integration/editors/triggered/ImportIedsPlugin.test.ts (hunk 1 above)
Comment (danyill) [reply test/integration/editors/triggered/ImportIedsPlugin.test.ts]: Sorry, different development environment and linter incorrectly configured (and earlier I had placed the test at the bottom).

Comment (danyill) [reply test/integration/editors/triggered/ImportIedsPlugin.test.ts]: Thanks for spotting the typo!

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Pull request merged**